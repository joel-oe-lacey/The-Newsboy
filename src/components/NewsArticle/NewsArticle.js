import React from 'react';
import './NewsArticle.scss';

const NewsArticle = ({article: {img, headline, description, url}}) => {
    return (<article className="news-article">
        <img src={img} alt={headline}></img>
        <section className="news-content">
            <h1>{headline}</h1>
            <p>{description}</p>
        </section>
        <button><a href={url}>Link to the article  -></a></button>
    </article>
    )}

export default NewsArticle;