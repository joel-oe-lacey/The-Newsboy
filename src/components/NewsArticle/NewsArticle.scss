.news-article {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    align-items: center;
    background-color: #86BBD8;
    border: 2px solid #33658A;
    margin: 1%;
    width: 48%;
    height: 70%;

    img {
      height: 55%;
      width: 90%;
      margin-top: 5%;
    }

    a {
      height: 8%;
      width: 100%;
      background-color: #F26419;
      font-size: 1.1em;
      font-weight: normal;
      text-decoration: none;
      padding-top: 1%;
      transition: all 0.4s ease 0s;

      &:hover {
        font-weight: bold;
        transition: all 0.4s ease 0s;
      }
    }
}

.news-content {
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    height: 40%;
    width: 90%;
}