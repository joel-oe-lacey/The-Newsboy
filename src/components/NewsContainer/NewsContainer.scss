.news-container {
    display: flex;
    flex-flow: row wrap;
    overflow: scroll;
    grid-area: news;
}