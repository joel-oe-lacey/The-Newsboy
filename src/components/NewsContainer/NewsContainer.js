import React from 'react';
import './NewsContainer.scss'
import NewsArticle from '../NewsArticle/NewsArticle';

const NewsContainer = (props) => {
    console.log('contProps', props)

    const newsArticles = props.articles.map(article => {
        return <NewsArticle key={article.id} article={article} />
    })

    return <section className="news-container">
        {newsArticles}
    </section>
}

export default NewsContainer;