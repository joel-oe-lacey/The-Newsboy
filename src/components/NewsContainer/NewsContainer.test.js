import React from 'react';
import { shallow } from 'enzyme';
import NewsContainer from './NewsContainer';
import NewsArticle from '../NewsArticle/NewsArticle';


describe('NewsContainer', () => {
    it('renders a container with passed articles', () => {
        const wrapper = shallow(<NewsContainer articles={[
            <NewsArticle
                key={1}
                article={{
                    img: 'www.imgTestUrl.com',
                    headline: 'Important Headline',
                    description: 'Important description text',
                    url: 'www.testRedirect.com'
            }} />,
            <NewsArticle
                key={2}
                article={{
                    img: 'www.imgTestUrl2.com',
                    headline: 'Other Headline',
                    description: 'Other description text',
                    url: 'www.testRedirect2.com'
            }} />,
        ]} />)

        expect(wrapper).toMatchSnapshot();
    })

    it('renders an error if no articles are available', () => {
        const wrapper = shallow(<NewsContainer articles={[]} />)

        expect(wrapper).toMatchSnapshot();
    })
})