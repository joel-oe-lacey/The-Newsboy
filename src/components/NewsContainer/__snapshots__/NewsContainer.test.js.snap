// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsContainer renders a container with passed articles 1`] = `
<section
  className="news-container"
>
  <NewsArticle
    article={
      <NewsArticle
        article={
          Object {
            "description": "The 24-foot-tall chicken skeleton towers over the people who pass through the main hall of the Denver Central Library. Skeletal wings extend to the side, and it looks downright prehistoric.",
            "headline": "Giant Chicken Skeleton In Denverâ€™s Central Library Stands For More Than Just Dinner",
            "id": 2,
            "img": "https://i0.wp.com/wp-cpr.s3.amazonaws.com/uploads/2019/09/cpr-swolf_black-cube-monumental_DSC8170.jpg?resize=2064,1376",
            "url": "https://www.cpr.org/2019/09/27/giant-chicken-skeleton-in-denvers-central-library-stands-for-more-than-just-dinner/",
          }
        }
      />
    }
    key="1"
  />
  <NewsArticle
    article={
      <NewsArticle
        article={
          Object {
            "description": "Car2Go, the popular car-sharing and rental company, will be leaving Denver at the end of the October, the company announced Friday.",
            "headline": "Car2Go leaving Denver amid company shake-up",
            "id": 3,
            "img": "https://www.denverpost.com/wp-content/uploads/2017/06/cd20carsharingdsc_8786.jpg?w=859",
            "url": "https://www.denverpost.com/2019/09/27/car2go-leaving-denver/",
          }
        }
      />
    }
    key="1"
  />
</section>
`;

exports[`NewsContainer renders an error if no articles are available 1`] = `
<section
  className="news-container"
>
  <h1
    className="no-results"
  >
    No matches found, please select your news type and search again.
  </h1>
</section>
`;
